scalding-core/src/main/scala/com/twitter/scalding/TimePathedSource.scala

```diff
@@ -31,6 +31,38 @@ object TimePathedSource {
       "%1$tm" -> Months(1)(tz), "%1$tY" -> Years(1)(tz))
       .find { unitDur: (String, Duration) => pattern.contains(unitDur._1) }
       .map(_._2)
+
+  /**
+   * Gives all paths in the given daterange with windows based on the provided duration.
+   */
+  def allPathsWithDuration(pattern: String, duration: Option[Duration], dateRange: DateRange, tz: TimeZone): Iterable[String] =
```

Comment:
I don't like passing an Option just to call .getOrElse(Nil). I'd rather require the duration here and have callers do that.